hack/generators/manifests/main.go

```diff
@@ -97,6 +123,13 @@ func processFile(path string, info os.FileInfo, err error) error {
 		return nil
 	}
 
+	// filter out YAML files for CRDs we specifically want to exclude from automatic generation
+	for _, excludedType := range excludedTypes {
+		if bytes.Contains(b, []byte(fmt.Sprintf("kind: %s", excludedType))) {
```

Comment:
This can bite us because the logic introduced here is susceptible to whitespace/quotation variability (think `kind:<space><space>"Xxx"`).

We could solve that by retrieving the `kind` in a robust way by deserializing into a `TypeMeta` and reading the kind from there.

I know that the current autogeneration will work with this implementation, and this program is "temporary", and a potential buggy execution will (probably) be caught in a code review including the results of the autogeneration. But I think that shortcuts of this kind set a problematic precedent for the codebase, so I vote in favor of fixing this.

Reply:
So it seems very unlikely to me that this would have ever come up since this script is only ever consuming YAML output from `controller-gen` which only includes a single space in its templating but since the change was minor and in order to make _sure_ it doesn't happen I fixed this in 11b9905e3302fcf2339a432ac9294c89b88208cc lmkwyt

Reply:
The fix looks great. I think that it's such small things, done over time, that gets us a codebase that's a pleasure to work with.